src/components/Home.jsx

```diff
@@ -1,5 +1,43 @@
+import { useContext, useEffect, useState } from "react";
+import { ActiveUserContext } from "../contexts/ActiveUserContext";
+import { getUsers } from "../api";
+
 const Home = () => {
-    return <h2>Home</h2>
+    const [chosenUser, setChosenUser] = useState('');
+    const [usersData, setUsersData] = useState([]);
+    const {activeUser, setActiveUser, isLoggedIn , setIsLoggedIn} = useContext(ActiveUserContext);
+
+    const loginHandler = (event) => {
+        event.preventDefault();
+        if (usersData.find(({username}) => username ===  chosenUser)) {
+            setActiveUser(chosenUser);
+            setIsLoggedIn(true);
+        }
+    }
+
+    const selectChangeHandler = (event) => {
+        const newUser = event.target.value;
+        setChosenUser(newUser);
+    }
+
+    useEffect(() => {
+        getUsers()
+        .then(data => setUsersData(data));
+    });
```

Comment:
This useEffect() has a missing dependency array, which is currently causing an infinite loop of re-renders and data fetching. It must have at least an empty array as a second parameter.

Reply:
My bad missed that one, fixed in latest commit.